mirascope/core/base/tool.py

```diff
@@ -23,9 +25,37 @@ class _CacheControl(TypedDict):
 
 
 class ToolConfig(TypedDict, total=False):
+    strict: bool
```

Comment:
@willbakst 
If this is a provider specific feature for OpenAI only, shouldn't `'strict` only be added to `OpenAIToolConfig` and not `ToolConfig`?

Reply:
However, this is fine if the design is to use a Tool for each provider.
I am concerned about the lack of clarity that this only applies to OpenAI.

Reply:
I can do that. I'll need to do the same for Anthropic since prompt caching is also provider-specific for Anthropic

Reply:
I see. looks good.

Reply:
yeah, I made `OpenAIToolConfig` and `AnthropicToolConfig` :)